do_function/monitor_snapshots/packages/monitor-snapshot/main/__main__.py

```diff
@@ -0,0 +1,136 @@
+import requests
+from xml.etree import ElementTree as ET
+import re
+import os
+from datetime import datetime, timezone, timedelta
+from slack import WebClient
+from slack.errors import SlackApiError
+import json
+
+# Define the DigitalOcean Spaces region and bucket.
+region = "fra1"
+bucket = "forest-snapshots"
+base_url = f"https://{region}.digitaloceanspaces.com/{bucket}"
+
+# Define the pattern for valid snapshot file names.
+pattern = r'([^_]+?)_snapshot_([^_]+?)_(\d{4}-\d{2}-\d{2})_height_(\d+).car(.zst)?$'
+
+# Define the folders that the script should scan for snapshots.
+folders = ["mainnet", "calibnet"]
+
+# Function to send alert messages to a Slack channel.
+def slack_alert(message_dict):
+    # Instantiate a Slack client with token from environment variables.
+    client = WebClient(token=os.environ['SLACK_TOKEN'])
+    CHANNEL_ID = 'C05BHMZ7GTT'
+    
+    # Format message as a JSON-like string for better readability.
+    message = f'```{json.dumps(message_dict, indent=4, ensure_ascii=False)}```'
+    
+    # Try sending message, catch and print any errors.
+    try:
+        response = client.chat_postMessage(channel=CHANNEL_ID, text=message)
+        print(f"Message sent. Response: {response['message']}")
+    except SlackApiError as e:
+        print(f"Slack API error: {e.response['error']}")
+
+# Function to get and return details of all snapshots from the server.
+def get_snapshots():
+    # Send a GET request to the server and parse the XML response.
+    response = requests.get(base_url)
+    root = ET.fromstring(response.text)
+    snapshots = {}
+    
+    # Iterate over each XML child element and extract snapshot details.
+    for child in root:
+        snapshot_dict = {}
+        snapshot_name = ''
+        
+        for snapshot in child:
+            if snapshot.tag.endswith('Key'):
+                snapshot_name = snapshot.text
+            elif snapshot.tag.endswith('Size'):
+                snapshot_dict['Size'] = int(snapshot.text)
+            elif snapshot.tag.endswith('LastModified'):
+                snapshot_dict['LastModified'] = datetime.strptime(snapshot.text, '%Y-%m-%dT%H:%M:%S.%fZ')
```

Comment:
Use the date field in the snapshot name rather than `LastModified`.

Reply:
Older snapshots may be uploaded at any time. Uploading a snapshot from last year shouldn't be mistaken for a new snapshot.

Reply:
How will that work?

Reply:
One day. If the last snapshot was for Monday and it's currently Wednesday, send an alert.